graph/node/node.go

```diff
@@ -0,0 +1,35 @@
+// Copyright © 2016 Asteris, LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package node
+
+// Node tracks the metadata associated with a node in the graph
+type Node struct {
+	ID string
+
+	value interface{}
+}
+
+// New creates a new node
+func New(id string, value interface{}) *Node {
```

Comment:
I've noticed that we have a general pattern of at each phase in the processing pipeline (`load`,`render`,`plan`,and `apply`) we create a new `*Node`, with the ID and value.  That works for now but it does require that we have knowlege that the id and value are the only relevant fields in the node, and that nothing earlier in the pipeline has added useful metadata.

Would it make sense to have a `func (n *Node) SetValue(newValue interface{}) {` that would set the value, or else a `func (n *Node) Transform(func (interface{}) (interface{}, error) ) (*Node, error)` that transforms the value inside of the node, so that later phases of the processing pipeline can update the inner value without losing accumulated metadata information?